### ISSUE TITLE
RuntimeError: generator raised StopIteration on PyQt5

### ISSUE THREAD

**Issue by zhou13** [davidhalter/jedi#1236]
````python
import os
import time
from jedi import Script
source = r"""
import numpy as np
import torch
from PyQt5 import QtGui, QtCore, QtWidgets

class MainWindow(QtWidgets.QMainWindow):
    def __init__(self):
        self.
""".strip()
lines = source.strip().split('\n')

ret = Script(source, len(lines), len(lines[-1]), None).completions()
````
yields
````
  File "/usr/lib/python3.7/site-packages/jedi/api/completion.py", line 43, in filter_names
    k = (new.name, new.complete)  # key
  File "/usr/lib/python3.7/site-packages/jedi/api/classes.py", line 434, in complete
    return self._complete(True)
  File "/usr/lib/python3.7/site-packages/jedi/api/classes.py", line 403, in _complete
    if self._name.api_type == 'param' and self._stack is not None:
  File "/usr/lib/python3.7/site-packages/jedi/evaluate/compiled/context.py", line 237, in api_type
    return next(iter(self.infer())).api_type
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "1.py", line 15, in <module>
    ret = Script(source, len(lines), len(lines[-1]), None).completions()
  File "/usr/lib/python3.7/site-packages/jedi/api/__init__.py", line 179, in completions
    completions = completion.completions()
  File "/usr/lib/python3.7/site-packages/jedi/api/completion.py", line 103, in completions
    return sorted(completions, key=lambda x: (x.name.startswith('__'),
RuntimeError: generator raised StopIteration
````

**Comment by ali-hardan** [comment]
I encountered this problem too, after upgrading jedi-vim using vim-plug.

**Comment by gad-o** [comment]
I face with the same problem. 
https://github.com/davidhalter/jedi/blob/f9ec989835cc1decfb478bce92edc3cec4f992fb/jedi/api/classes.py#L409
This line was changed in this commit
https://github.com/davidhalter/jedi/commit/f9ec989835cc1decfb478bce92edc3cec4f992fb#diff-309625ee9f955d6f72e03bed5c326661L409
Probleblem resoveled when replace this line with old line for me

**Comment by randrej** [comment]
I have the same problem when using coc.nvim, coc-pyls and pyls. Python 3.7.3, jedi 0.13.3

**Comment by davidhalter** [comment]
Fixed by #1352. Thanks for the report!
